lib/Rpc/RpcServer.cpp

```diff
@@ -1123,6 +1135,90 @@ bool RpcServer::on_get_transactions(
     return true;
 }
 
+bool RpcServer::onGetTransactionsByHeights(
+        const COMMAND_RPC_GET_TRANSACTIONS_BY_HEIGHTS::request &req,
+        COMMAND_RPC_GET_TRANSACTIONS_BY_HEIGHTS::response &res)
+{
+    try {
+        std::vector<Crypto::Hash> vh;
+
+        size_t additor = 100; // possible to increase
+        for (size_t i = req.startBlock; i <= req.startBlock + additor; i++) {
```

Comment:
We can calc the upper border instead of throwing,  smth. like:
```suggestion
        uint32_t upperBorder = std::min(req.startBlock + additor, m_core.get_current_blockchain_height());
        for (size_t i = req.startBlock; i <= upperBorder; i++) {
```
In this case we don't need to throw for NULL_HASH